src/ring_theory/trace.lean

```diff
@@ -97,6 +97,38 @@ begin
   { simp [trace_eq_zero_of_not_exists_basis K H, finrank_eq_zero_of_not_exists_basis_finset H] }
 end
 
+lemma trace_trace_of_basis [algebra S T] [is_scalar_tower R S T]
+  {ι κ : Type*} [fintype ι] [fintype κ]
+  (b : basis ι R S) (c : basis κ S T) (x : T) :
+  trace R S (trace S T x) = trace R T x :=
```

Comment:
Is this formulation equivalent?
```suggestion
lemma trace_trace_of_basis [algebra S T] [is_scalar_tower R S T]
  [finite_dimensional R S] [finite_dimensional S T] (x : T) :
  trace R S (trace S T x) = trace R T x :=
```
It feels like "has a finite basis" is something that we ought to capture via typeclasses, but I don't know for sure whether the meaning of `finite_dimensional` is close enough to that.

Reply:
See `trace_trace` a few lines below :)

"A finite basis exists" and `finite_dimensional` is not quite the same thing currently, since `finite_dimensional` is defined only for vector spaces (easy to solve) as `is_noetherian` (which I'm pretty sure neither implies nor follows from "a finite basis exists", even if we assume `module.free`)

Reply:
See e.g. this discussion on Zulip: https://leanprover.zulipchat.com/#narrow/stream/267928-condensed-mathematics/topic/free.20module